https.go

```diff
@@ -555,8 +555,10 @@ func httpsProxyFromEnv(reqURL *url.URL) (string, error) {
 
 	service := proxyURL.Port()
 	if service == "" {
+		// TODO I am very confused by this line of code
```

Comment:
Looking at the docs for `net.Dial` where the return value from the function is eventually consumed:

```
// For TCP and UDP networks, the address has the form "host:port".
// The host must be a literal IP address, or a host name that can be
// resolved to IP addresses.
// The port must be a literal port number or a service name.
// Examples:
//	Dial("tcp", "golang.org:http")
//	Dial("tcp", "192.0.2.1:http")
```

If I had to guess, this is just a shorthand way to attach the correct port number without checking the scheme in an if statement before attaching the port literal.

Reply:
Ah, interesting!